tools: add version to maintaining-dependencies.md dependencies list and automate the update

fasenderos opened the pull request:
Add version to `doc/contributing/maintaining/maintaining-dependencies.md` dependencies list and automate the update
Refs: https://github.com/nodejs/security-wg/issues/973

Most of the duplicated code in all the dependencies updaters has been moved in the `utils.sh` file

This is the list of all the dependencies:
- [x] acorn
- [x] ada
- [x] base64
- [x] brotli
- [x] c-ares
- [x] cjs-module-lexer
- [ ] corepack
- [x] googletest
- [x] histogram
- [x] icu-small
- [x] llhttp
- [x] minimatch
- [x] nghttp2
- [x] ngtcp2
- [x] npm
- [x] openssl
- [x] postject
- [x] simdutf
- [x] undici
- [x] uv
- [x] uvwasi
- [x] V8
- [x] zlib

fasenderos commented:
1. For `corepack` and `histogram` I was not able to locate the update script.
2. The update scripts of `postject`, `undici` and `V8` it seems they differ from the others because they don't commit any changes, so probably the dependency update is done in another way. Any hint?
3. The `zlib` update script is currently under review of [another PR](https://github.com/nodejs/node/pull/48054) that will change the update routine

tniessen reviewed (commented):
I know that the existing update scripts are already not pretty, but I still don't think we should add duplicate code to all of them. Could you please DRY the logic?

fasenderos commented:
@tniessen I'm agree with you. I started recently to collaborate on node and I wasn't sure if I could (or not) create new files. Anyway if [this refactor](https://github.com/nodejs/node/pull/48081/commits/a873cadcbbe8893636713d1330e6f20dafa889cb) is ok, than I can continue with the other dependencies.

RaisinTen commented on doc/contributing/maintaining/maintaining-dependencies.md:
```diff
@@ -268,77 +281,72 @@ the main openssl/openssl releases with the addition of APIs to support
 See [maintaining-openssl][] for more informations.
 
-### postject
+### postject 1.0.0-alpha.5
```
Postject has recently been upgraded to 1.0.0-alpha.6, so this would need an update

marco-ippolito commented:
the version of undici is currently `5.22.1` https://github.com/nodejs/node/blob/main/deps/undici/src/package.json
you can update it with `NEW_VERSION` https://github.com/nodejs/node/blob/main/tools/dep_updaters/update-undici.sh

fasenderos commented:
@marco-ippolito I have added version to `undici`, `postject` and `V8` and refactored all these scripts by moving most of the duplicated code in the `utils.sh` file.

marco-ippolito commented:
pr for histogram dep_updater : https://github.com/nodejs/node/pull/48171

fasenderos commented:
@marco-ippolito added version for `histogram`.

targos commented on doc/contributing/maintaining/maintaining-dependencies.md:
```diff
@@ -8,29 +8,30 @@ directories to create the Node.js binaries.
+* [simdutf 3.2.9][]
+* [undici 5.22.1][]
+* [uvwasi 0.0.16][]
+* [V8 11.3.244.8][]
```
Someone please send a PR to https://github.com/nodejs/node-core-utils/tree/main/lib/update-v8 to automatically change this markdown file during major V8 updates.

fasenderos replied:
Here is the PR https://github.com/nodejs/node-core-utils/pull/699
